server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/MessageSender.java

```diff
@@ -0,0 +1,70 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.methods;
+
+import java.io.IOException;
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.mail.MessagingException;
+
+import org.apache.james.jmap.model.CreationMessageId;
+import org.apache.james.jmap.model.Message;
+import org.apache.james.jmap.model.MessageFactory;
+import org.apache.james.jmap.send.MailFactory;
+import org.apache.james.jmap.send.MailMetadata;
+import org.apache.james.jmap.send.MailSpool;
+import org.apache.james.lifecycle.api.LifecycleUtil;
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.exception.MailboxException;
+import org.apache.mailet.Mail;
+
+import com.google.common.collect.ImmutableList;
+
+public class MessageSender {
+    private final MailSpool mailSpool;
+    private final MailFactory mailFactory;
+
+    @Inject
+    public MessageSender(MailSpool mailSpool, MailFactory mailFactory) {
+        this.mailSpool = mailSpool;
+        this.mailFactory = mailFactory;
+    }
+
+    public void sendMessage(Message jmapMessage,
```

Comment:
how are the two first argument related ?

Reply:
One is generated from the other. 

You have 3 related things:
 - CreationMessage
 - MessageMetadata: once you saved it in store
 - JMAP message is used only for the enveloppe. 

We shoudl have the MailFactory taking an Enveloppe instead of a full JMAP message.